templates/github/.github/workflows/release.yml.j2

```diff
@@ -127,56 +103,55 @@ jobs:
           {{ set_env_vars() | indent(10) }}
         shell: bash
 
-      {{ run_script(name="Before Script", file="before_script.sh", extra_env={"REDIS_DISABLED": "${{ contains('%s', matrix.env.TEST) }}" % (disabled_redis_runners | default(["TILT"]) | join(' '))}) | indent(6) }}
-
-      {{ set_secrets() | indent(6) }}
-
-      {{ run_script(name="Install Python client", file="install_python_client.sh", withenv=False) | indent(6) }}
-
-      {{ run_script(name="Install Ruby client", file="install_ruby_client.sh", withenv=False, condition="${{ env.TEST == 'bindings' || env.TEST == 'generate-bindings' }}") | indent(6) }}
-
       - name: Additional before_script
         run: {{ "${{ github.event.inputs.before_script }}" }}
         shell: bash
 
-      {{ run_script(name="Script", file="script.sh", condition="${{ env.TEST != 'generate-bindings' }}") | indent(6) }}
+      {{ run_script(name="Install Python client", file="install_python_client.sh", withenv=False) | indent(6) }}
+
+      {%- if deploy_client_to_rubygems %}
+      {{ run_script(name="Install Ruby client", file="install_ruby_client.sh", withenv=False) | indent(6) }}
+      {%- endif %}
 
       - name: Upload python client packages
-        if: {{ "${{ env.TEST == 'bindings' || env.TEST == 'generate-bindings' }}" }}
         uses: actions/upload-artifact@v3
         with:
           name: python-client.tar
           path: python-client.tar
 
       - name: Upload python client docs
-        if: {{ "${{ env.TEST == 'bindings' || env.TEST == 'generate-bindings' }}" }}
         uses: actions/upload-artifact@v3
         with:
           name: python-client-docs.tar
           path: python-client-docs.tar
 
+      {%- if deploy_client_to_rubygems %}
       - name: Upload ruby client packages
-        if: {{ "${{ env.TEST == 'bindings' || env.TEST == 'generate-bindings' }}" }}
         uses: actions/upload-artifact@v3
         with:
           name: ruby-client.tar
           path: ruby-client.tar
+      {%- endif %}
+
+      - name: Build docs
+        run: |
+          pip install towncrier==19.9.0
+          towncrier --yes --version 4.0.0.ci
+          make -C docs/ PULP_URL="{{ pulp_scheme }}://pulp" diagrams html
+          tar -cvf docs/docs.tar docs/_build
 
-      {%- if docs_test %}
```

Comment:
This is a plugin-template switch. I think for plugins that do not have/build docs we need to keep this, or consolidate with `publish_docs_to_pulpprojectdotorg`.